integrationTests/state/stateTrie/stateTrie_test.go

```diff
@@ -827,6 +827,158 @@ func TestAccountsDB_RevertDataStepByStepWithCommitsAccountDataShouldWork(t *test
 	require.Equal(t, hrRoot2, hrRoot2Rev)
 }
 
+func TestAccountsDB_RevertDataStepByStepWithCommitsAccountDataWithMigratedCode(t *testing.T) {
+	t.Parallel()
+
+	t.Run("account with migrated code leaf activation, should code data to storage directly", func(t *testing.T) {
```

Comment:
some words are missing, this does not make sense

Reply:
You do not need 2 accounts for this test.

Reply:
Can add other tests: 

- You have 2 accounts that have the same code, so the code entry is referenced 2 times. Migrate one and revert it
- 1 account has code. Migrate it and then revert the migration

Reply:
Also, do these tests by using the built in func. Check `integrationTests/vm/txsFee/migrateDataTrie_test.go` and `integrationTests/vm/txsFee/dynamicGasCost_test.go` for examples